modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemLocalServiceImpl.java

```diff
@@ -2623,6 +2640,242 @@ private void _validate(
 		}
 	}
 
+	private String _validateJSON(
+			CPDefinition cpDefinition, CPInstance cpInstance, String json)
+		throws PortalException {
+
+		String sanitizedJSON = json;
+
+		if (cpDefinition != null) {
+			int optionRelCount =
+				_cpDefinitionOptionRelLocalService.
+					getCPDefinitionOptionRelsCount(
+						cpDefinition.getCPDefinitionId());
+
+			if (optionRelCount > 0) {
+				JSONArray optionJSONArray = CPJSONUtil.toJSONArray(json);
+
+				for (CPDefinitionOptionRel cpDefinitionOptionRel :
+						cpDefinition.getCPDefinitionOptionRels()) {
+
+					if (cpDefinitionOptionRel.isSkuContributor()) {
+						if (cpInstance == null) {
+							throw new NoSuchCPInstanceException();
```

Comment:
Double check to make sure you are properly handling this exception on both headless and MVCActionCommands.
Above comment will apply to all of your implementations so please check for all of them.

Reply:
This check is also not necessary no?
`CPInstance cpInstance = _cpInstanceLocalService.getCPInstance(cpInstanceId);` will always throw an exception.

Reply:
Reasoning for check was due to same reasoning as other checks, but I can omit if necessary.